index.js

```diff
@@ -37,18 +87,42 @@
  *
  * And returns the first and last name concatenated together (i.e. 'Bob Evans').
  */
+ function getFullName(firstName, lastName) {
+ 	var fullName = {
+ 		firstName: 'Bob',
+	 	lastName: 'Evans'
+	 };
+	 console.log(fullName.firstName + ' ' + fullName.lastName);
+	 return fullName.firstName + ' ' + fullName.lastName;
+ }
+ getFullName();
 
 
 /**
  * Q6: Write a function called `reverseString` that has a String parameter and returns
  * the String reversed. Hint: look up `.split()`
  */
-
+ function reverseString(string) {
+ 	return string.split('').reverse().join('');
+ }
+console.log(reverseString('palindrome'));
 
 /**
  * Q7: Write a function called `maxArray` that has one Array of Numbers parameter and returns
  * the largest number in the Array. Use a `forEach` loop.
  */
+ function maxArray() {
+ 	var numbersArray = [1, 14, 956, 4, 8, 2];
+ 	var largestNumber = 0;
+ 	var compare = function(item) {
+ 		if (largestNumber > item) {
+ 			largestNumber = item;
+ 		}
+ 	};
+ 	numbersArray.forEach(compare);
+ 	console.log(largestNumber);
+ }
```

Comment:
This function is nearly perfect, there is just a few things that need to be modified:

1. Rather than declaring the array inside of the function, we want to accept it as input to the function so that it can be used for any array:

```js
function maxArray(numbersArray) {
    var largestNumber = 0;
    var compare = function(item) {
        if (largestNumber > item) {
            largestNumber = item;
        }
    };
    numbersArray.forEach(compare);
    console.log(largestNumber);
}
```

2. What we want to check in our loop is whether or not the number we are looking at now is bigger than any number we've seen before (the `largestNumber`). What you have done is the opposite: is the largestNumber larger than this number. We need to flip that inequality:

```js
function maxArray(numbersArray) {
    var largestNumber = 0;
    var compare = function(item) {
        if (largestNumber < item) {
            largestNumber = item;
        }
    };
    numbersArray.forEach(compare);
    console.log(largestNumber);
}
```

3. Since we are starting the `largestNumber` variable at `0`, what happens if all the numbers in the array are negative? Well, since none of them are greater than 0, 0 will never be usurped as the largestNumber, even though it is not in the array. A better approach would be to initialize the largestNumber as the first element in the array:

```js
function maxArray(numbersArray) {
    var largestNumber = numbersArray[0];
    var compare = function(item) {
        if (largestNumber < item) {
            largestNumber = item;
        }
    };
    numbersArray.forEach(compare);
    console.log(largestNumber);
}
```

4. The question asks that we return the largest number in the array rather than console logging it:

```js
function maxArray(numbersArray) {
    var largestNumber = numbersArray[0];
    var compare = function(item) {
        if (largestNumber < item) {
            largestNumber = item;
        }
    };
    numbersArray.forEach(compare);
    return largestNumber;
}
```

5. Finally, though this doesn't affect the functionality at all, you could pass the `compare` function directly into the `forEach` call, rather than storing it in an intermediate variable:

```js
function maxArray(numbersArray) {
    var largestNumber = numbersArray[0];

    numbersArray.forEach(function(item) {
        if (largestNumber < item) {
            largestNumber = item;
        }
    });
    return largestNumber;
}
```

Now, we could call our new function:

```js
var numbers = [0, 4, 5, 6, 7, 8, -1, -4];
var largestInArray = maxArray(numbers);

console.log(largestInArray);
// 8
```